Modules/FindCUDA.cmake

```diff
@@ -683,6 +684,24 @@ find_path(CUDA_TOOLKIT_INCLUDE
 find_path(CUDA_TOOLKIT_INCLUDE device_functions.h)
 mark_as_advanced(CUDA_TOOLKIT_INCLUDE)
 
+if (CUDA_VERSION VERSION_GREATER "7.0")
+  set(cuda_fp16_found ${CUDA_TOOLKIT_INCLUDE})
+else()
+  find_path( cuda_fp16_found
+    cuda_fp16.h # Header included in toolkit
+    PATHS ${CUDA_TOOLKIT_INCLUDE}
+    ENV CUDA_PATH
+    ENV CUDA_INC_PATH
+    NO_DEFAULT_PATH
+    )
```

Comment:
I agree.  We could use:

if(EXISTS "${CUDA_TOOLKIT_INCLUDE}/cuda_fp16.h") to set the variable.  It doesn't need to be in the cache.